modules/tensor_mechanics/doc/content/modules/tensor_mechanics/1d_elastic_waves.md

```diff
@@ -0,0 +1,104 @@
+
+Now, equation (2) and the associated BCs. and can be written as,
+\begin{equation}
+     -\nabla \cdot \sigma- \omega^2 u = 0 \label{eq4}
+\end{equation}
+
+\begin{equation}
+    \nabla u= ik \, u \, \qquad at \, x=0,
+    \\
+    u= A \, cos(2 \pi f t) \, \qquad at \, x=L,\label{eq5}
+\end{equation}
+
+Transforming equation (4) and the associated boundary conditions (5) in time to the frequency domain, we obtain:
+\begin{equation}
+     -\nabla \cdot \hat \sigma- \omega^2  U = 0 \label{eq6}
+\end{equation}
+with the following boundary conditions:
+\begin{equation}
+    \nabla U= ik \, U \, \qquad at \, x=0,
+    \\
+U= 0.5 \, \qquad at \, x=L,\label{eq7}
+\end{equation}
+
+where $\hat \sigma = \nabla U$ with $U(x,\omega)$ being the Fourier transform of $u(x,t)$.
+
+!alert note title=Fourier Transfrom of Boundary Conditions
+Not only the pde is transformed to the frequency domain, but also, we must transform all B.Cs as well. Transfroming $A cos(2\pi f t)$ into the frquency domain will result in an impulse at frequency = $f$ with amplitude of A.
+
+When transform the time dependent variable $u$ to the frequency domain, the time-dependency will be omitted. The result is a nother variable $U$ that is frequency dependent. In general, $U$ is complex-valued function/variable in the form of $U_r+iU_i$, where $U_r$ and $U_i$ are the real and the imaginary part of $U$.
+
+At this stage, we split the complex sytem of equations into two $\textbf{real}$ systems, which yields two real-valued systems that live on the same mesh. The strong form PDEs becomes as follows:
+
+\begin{equation}
+    -\nabla \cdot \hat \sigma_r- \omega^2  U_r = 0
+    \\
+    -\nabla \cdot \hat \sigma_i- \omega^2  U_i = 0 \label{eq8}
+\end{equation}
+with the following boundary conditions:
+\begin{equation}
+   \nabla U_r= -k U_i \quad ,\nabla U_i= k U_r\, \qquad at \, x=0,\\
+   U_r=U_i=0 \, \qquad at \, x=L, \label{eq9}
+\end{equation}
+
+Note that this decomposition is exact and there is no infromation missed while decompisition. Also note that, as a result of the radiation BCs on left, $U_r$ and $U_i$ are coupled, hence the two systems in (8).
+
+The following is how we define the kernels and the BCs. of this problem,
+
+!listing examples/wave_propagation/1D_elastic_wave_propagation.i block=Kernels id=listing1
+
+ and here is how we define the BCs. of this problem,
+
+!listing examples/wave_propagation/1D_elastic_wave_propagation.i block=BCs id=listing2
+
+!alert note
+$u$ is a function of time and space $(x,t)$, whereas $U$ is function of space and frequency $(x,\omega)$.
+
+The moose solution for this problem is shown below, see[fig_cook2].
+
+!media media/tensor_mechanics/wavefieldandprofile.png
+        style=width:40%;
+        id=fig_cook2;
```

Comment:
It is still failing.  Use something like this all on one line and put id last:

```!media stochastic_tools/surrogates/combined/diff_trans_2d/geometry.png style=width:45%;float:left id=geometry```

Reply:
Done. I think I found where it was failing. Now this should pass, hopefully.